src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -763,53 +722,12 @@ public static string Join(string separator, params object[] values)
         }
 
         [ComVisible(false)]
-        public static String Join<T>(String separator, IEnumerable<T> values)
+        public unsafe static string Join<T>(string separator, IEnumerable<T> values)
         {
-            if (values == null)
-                throw new ArgumentNullException("values");
-            Contract.Ensures(Contract.Result<String>() != null);
-            Contract.EndContractBlock();
-
-            using (IEnumerator<T> en = values.GetEnumerator())
+            fixed (char* pSeparator = &separator.m_firstChar)
```

Comment:
@justinvp Then `pSeparator` will be initialized to null. This is technically implementation-defined behavior of Roslyn, but [it looks like they're planning to update the spec](https://github.com/dotnet/roslyn/issues/6707) sometime.

On hindsight, the real issue here looks like `separator.Length` will nullref if it is null... I'll just change this to `separator = separator ?? string.Empty`.